plasma_framework/contracts/src/exits/payment/controllers/PaymentProcessInFlightExit.sol

```diff
@@ -0,0 +1,224 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "../PaymentExitDataModel.sol";
+import "../PaymentInFlightExitModelUtils.sol";
+import "../../../vaults/EthVault.sol";
+import "../../../vaults/Erc20Vault.sol";
+import "../../../framework/PlasmaFramework.sol";
+
+library PaymentProcessInFlightExit {
+    using PaymentInFlightExitModelUtils for PaymentExitDataModel.InFlightExit;
+
+    uint8 public constant MAX_INPUT_NUM = 4;
+    uint8 public constant MAX_OUTPUT_NUM = 4;
+
+    struct Controller {
+        PlasmaFramework framework;
+        EthVault ethVault;
```

Comment:
For that idea to be useful it needs huge changes I think. We'd need to change how the transaction output should look like. Otherwise we will just be moving our logic of determine vault from here to the step of piggyback.